pandas/_libs/window/aggregations.pyx

```diff
@@ -1379,83 +1079,15 @@ def roll_generic_fixed(object obj,
         if not arr.flags.c_contiguous:
             arr = arr.copy('C')
 
-    counts = roll_sum_fixed(np.concatenate([np.isfinite(arr).astype(float),
-                                            np.array([0.] * offset)]),
-                            start, end, minp, win)[offset:]
+    counts = roll_sum(np.isfinite(arr).astype(float), start, end, minp)
 
     output = np.empty(N, dtype=float)
 
-    if not raw:
-        # series
-        for i in range(N):
-            if counts[i] >= minp:
-                sl = slice(int_max(i + offset - win + 1, 0),
-                           int_min(i + offset + 1, N))
-                output[i] = func(obj.iloc[sl], *args, **kwargs)
-            else:
-                output[i] = NaN
-
-    else:
-
-        # truncated windows at the beginning, through first full-length window
-        for i in range((int_min(win, N) - offset)):
-            if counts[i] >= minp:
-                output[i] = func(arr[0: (i + offset + 1)], *args, **kwargs)
-            else:
-                output[i] = NaN
-
-        # remaining full-length windows
-        for j, i in enumerate(range((win - offset), (N - offset)), 1):
-            if counts[i] >= minp:
-                output[i] = func(arr[j:j + win], *args, **kwargs)
-            else:
-                output[i] = NaN
-
-        # truncated windows at the end
-        for i in range(int_max(N - offset, 0), N):
-            if counts[i] >= minp:
-                output[i] = func(arr[int_max(i + offset - win + 1, 0): N],
-                                 *args,
-                                 **kwargs)
-            else:
-                output[i] = NaN
-
-    return output
+    #if offset != 0:
+    #    raise ValueError("unable to roll_generic with a non-zero offset")
```

Comment:
leftover?

Reply:
Correct, still attempting to figure out how `center` is handled here with `rolling.apply`